qiskit_experiments/visualization/plotters/curve_plotter.py

```diff
@@ -0,0 +1,137 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""Plotter for curve-fits, specifically from :class:`CurveAnalysis`."""
+from typing import List
+
+from qiskit_experiments.framework import Options
+
+from .base_plotter import BasePlotter
+
+
+class CurvePlotter(BasePlotter):
+    """A plotter class to plot results from :class:`CurveAnalysis`.
+
+    :class:`CurvePlotter` plots results from curve-fits, which includes:
+        Raw results as a scatter plot.
+        Processed results with standard-deviations/confidence intervals.
+        Interpolated fit-results from the curve analysis.
+        Confidence interval for the fit-results.
+        A report on the performance of the fit.
+    """
+
+    @classmethod
+    def expected_series_data_keys(cls) -> List[str]:
+        """Returns the expected series data-keys supported by this plotter.
+
+        Data Keys:
+            x: X-values for raw results.
+            y: Y-values for raw results. Goes with ``x``.
+            x_formatted: X-values for processed results.
+            y_formatted: Y-values for processed results. Goes with ``x_formatted``.
+            y_formatted_err: Error in ``y_formatted``, to be plotted as error-bars.
+            x_interp: Interpolated X-values for a curve-fit.
+            y_mean: Y-values corresponding to the fit for ``x_interp`` X-values.
+            sigmas: The standard-deviations of the fit for each X-value in ``x_interp``.
+                This data-key relates to the option ``plot_sigma``.
+        """
+        return [
+            "x",
+            "y",
+            "x_formatted",
+            "y_formatted",
+            "y_formatted_err",
+            "x_interp",
+            "y_mean",
+            "sigmas",
```

Comment:
```suggestion
            "y_interp",
            "y_interp_err",
```
to align with `_formatted`. I think changing names are okay because the logic to generate figure is now accessed only internally.

Reply:
Addressed in 9a909f5 and fdf1c23.